src/App.js

```diff
@@ -0,0 +1,101 @@
+import { useEffect, useState } from 'react'
+import { Routes, Route, createSearchParams, useSearchParams, useNavigate } from "react-router-dom"
+import { useDispatch, useSelector } from 'react-redux'
+import 'reactjs-popup/dist/index.css'
+import { fetchMovies } from './data/moviesSlice'
+import { ENDPOINT_SEARCH, ENDPOINT_DISCOVER, ENDPOINT, API_KEY } from './constants'
+import Header from './components/Header'
+import Movies from './components/Movies'
+import Starred from './components/Starred'
+import WatchLater from './components/WatchLater'
+import YouTubePlayer from './components/YoutubePlayer'
+import './app.scss'
+
+const App = () => {
+
+  const state = useSelector((state) => state)
+  const { movies } = state  
```

Comment:
To optimize performance and minimize unnecessary re-renders, it's recommended to subscribe specifically to the part of the state required by this component.  The original implementation subscribes to the entire state, potentially causing unnecessary re-renders whenever any part of the state changes. By targeting only the 'movies' state slice with 'useSelector', the component will re-render solely upon changes in the 'movies' part of the state, improving performance.

Instead of:
```
const state = useSelector((state) => state)
const { movies } = state
```

Consider this approach:
 ```
 const movies = useSelector((state) => state.movies)
```